src/devices/esp8266.c

```diff
@@ -123,6 +123,32 @@ static void ipd_urc_cb(char *msg)
 		state.hooks.data_received_cb(chan_id, len, data);
 }
 
+static void wifi_action_callback(const char* msg)
+{
+	/* Check for the hook */
+	if (!state.hooks.client_state_changed_cb)
+		return;
+
+	enum client_action action;
+	/* Look at our 5th character to determin our action */
+	switch (msg[5]) {
```

Comment:
Then the use of this method is being done incorrectly.  I only bounds check public methods or methods that are callbacks.  In this case since this is internal I do not bounds check.